apps/stable_diffusion/src/models/model_wrappers.py

```diff
@@ -549,138 +565,161 @@ def process_custom_vae(self):
-        print("Inferring base model configuration.")
         for model_id in base_models:
             try:
-                if need_vae_encode:
-                    compiled_clip, compiled_unet, compiled_vae, compiled_vae_encode = self.compile_all(model_id, need_vae_encode, need_stencil)
-                elif need_stencil:
-                    compiled_clip, compiled_unet, compiled_vae, compiled_controlnet = self.compile_all(model_id, need_vae_encode, need_stencil)
-                else:
-                    compiled_clip, compiled_unet, compiled_vae = self.compile_all(model_id, need_vae_encode, need_stencil)
+                compiled_vae_encode = self.compile_model(model_id, "vae_encode")
             except Exception as e:
                 print(e)
                 print("Retrying with a different base model configuration")
                 continue
             # -- Once a successful compilation has taken place we'd want to store
             #    the base model's configuration inferred.
-            fetch_and_update_base_model_id(model_to_run, model_id)
+            fetch_and_update_base_model_id(self.model_to_run, model_id)
             # This is done just because in main.py we are basing the choice of tokenizer and scheduler
             # on `args.hf_model_id`. Since now, we don't maintain 1:1 mapping of variants and the base
             # model and rely on retrying method to find the input configuration, we should also update
             # the knowledge of base model id accordingly into `args.hf_model_id`.
             if args.ckpt_loc != "":
                 args.hf_model_id = model_id
-            if need_vae_encode:
-                return (
-                    compiled_clip,
-                    compiled_unet,
-                    compiled_vae,
-                    compiled_vae_encode,
-                )
-            if need_stencil:
-                return (
-                    compiled_clip,
-                    compiled_unet,
-                    compiled_vae,
-                    compiled_controlnet,
-                )
-            return compiled_clip, compiled_unet, compiled_vae
+            return compiled_vae_encode
+        sys.exit(
+            "Cannot compile the model. Please create an issue with the detailed log at https://github.com/nod-ai/SHARK/issues"
+        )
+
+
+    def clip(self):
+        vmfb = fetch_vmfb("clip", self.model_name["clip"], self.precision)
+        if vmfb:
+            return vmfb
+        if self.base_model_fetched != "":
+            return self.compile_model(self.base_model_fetched, "clip")
+        for model_id in base_models:
+            try:
+                compiled_clip = self.compile_model(model_id, "clip")
+            except Exception as e:
+                print(e)
+                print("Retrying with a different base model configuration")
+                continue
+            fetch_and_update_base_model_id(self.model_to_run, model_id)
```

Comment:
I see instances like these all over.
This is completely wrong for 2 reasons :-
1. You're performing the "retry" for each individual models. So, theoretically, if there were 'N' retries previously - this would now involve a total of ~3N-4N retries.
2. For each model you're invoking `fetch_and_update_base_model_id`! This is so wrong which is apparent in the following scenario :-
 Say, UNET for `v1.5` compiles first then we'd map `model_to_run` with "CompVis v1.4". After this when VAE compiles it'd end up replacing "v2.1" with, say, "upscaler" base ID. And that's wrong because now we're in the state where "v1.5" is being mapped to "v2.1", instead of "v1.4" in variants.json, and therefore in the next invocation, if someone uses `--clear_all`, because of https://github.com/nod-ai/SHARK/blob/e9f45d6b9d1712f2a5d3bb3944f6b1dc5536f967/apps/stable_diffusion/src/models/model_wrappers.py#L166 the system would crash because you're trying to compile "v1.5" Unet with "v2.1"'s input configuration due to : https://github.com/nod-ai/SHARK/blob/e9f45d6b9d1712f2a5d3bb3944f6b1dc5536f967/apps/stable_diffusion/src/models/model_wrappers.py#L633-L634